File: Java/Arrays/src/rotateArray.java
/*
 * So, for example,


A : [1 2 3 4 5 6]
B : 1

The output :

[2 3 4 5 6 1]
 */
import java.util.*;
public class rotateArray {
	public ArrayList<Integer> rotate(ArrayList<Integer> A, int B) {
		ArrayList<Integer> ret = new ArrayList<Integer>();
		
		if (A.size() == 1) {
		    ret.add(A.get(0));
		}
		
		else {
		    
		    int x = 0;
		    
    		for (int i = 0; i < A.size(); i++) {
                
                if ((x + B) >= A.size()) {
                     x = (x+B)%A.size() - B;
                }
               
            	ret.add(A.get(x + B));
            	x++;

    		}		    
		}
		return ret;
	}
}

File: Python/Arrays/rotateArray.py
class Solution:
    # @param a : list of integers
    # @param b : integer
    # @return a list of integers
    def rotateArray(self, a, b):
        ret = []
        
        for i in xrange(len(a)):
            ret.append(a[(i + b)%len(a)])

                
        return ret


test = Solution()
result = test.rotateArray([1,2,3,4,5,6,7,8,9,10], 12)
print result



